### DISCUSSION TITLE
"FWscHIST" custom compset vs. out-of-the-box "FHIST" compset

### DISCUSSION THREAD

**Discussion by adamrher** [NCAR/amwg_dev#66] [TF1: Unified physics for L58/L93]
@cecilehannay I was wondering if we could get some amwg diagnostics showing the impact of specified chemistry in L58. I see in your campaign storage directory you have:

f.e21.FHIST.ne30_L48_BL10_cam6_3_019_plus_CESM2.2.001_zm2.hf
f.e21.FWscHIST.ne30_L48_BL10_cam6_3_019_plus_CESM2.2.001_zm2.hf

Which I think would address differences due to specified chemistry.

Thanks!
Adam

**Comment by cecilehannay** [comment]
Hi @adamrher,

The diags are here:
https://webext.cgd.ucar.edu/FHIST/f.e21.FHIST.ne30_L48_BL10_cam6_3_019_plus_CESM2.2.001_zm2.hf/atm/

Indeed, it was the purpose of that simulation to address the differences between FHIST and FWscHIST in the **low top model.**  

At the time, we decided that the climate of FHIST and FWscHIST were similar enough and the extra cost was small enough (10-20%) to justify to use FWscHIST in both the low top and high models. But we can revisit that.

**Comment by swrneale** [comment]
There are certainly consistent +1K differences in the tropical tropopause temp.
(https://webext.cgd.ucar.edu/FHIST/f.e21.FHIST.ne30_L48_BL10_cam6_3_019_plus_CESM2.2.001_zm2.hf/atm/f.e21.FHIST.ne30_L48_BL10_cam6_3_019_plus_CESM2.2.001_zm2.hf.1980_1989-f.e21.FWscHIST.ne30_L48_BL10_cam6_3_019_plus_CESM2.2.001_zm2.hf.1980_1989/set5_6/set5_ANN_TTRP_TROP_c.png)

I wonder if we could do the WACCM version of the diagnsotics?

**Comment by JulioTBacmeister** [comment]
What is the impact on the water vapor bias?

**Reply by adamrher** [reply]
Since set4 doesn't work in the amwg diags ... Ill make some zonal height difference plots of T and Q

**Comment by islasimpson** [comment]
I can  make the tape recorder plot

**Comment by swrneale** [comment]
Cecile, I'm pretty sure I fixed set4 in the past (my script works). Can you point me to your version of the code and then I can compare?

**Reply by cecilehannay** [reply]
I use: /glade/p/cesm/postprocessing/atm_diag

**Comment by adamrher** [comment]
Because both these runs use ZM2 from the "cam6_3_019_plus_CESM2.2.001" runs, I added both of these runs to the bottom row of the Q plots [here](https://webext.cgd.ucar.edu/FSCAM/FWscHIST/L32_REF/02042022/temp_dzonal.Q.logp.pdf). In those runs, the UT is a bit drier than our current L58+zm2 control. But FHIST has a bit moister lower stratosphere.

Note I also have 5 years of the new PUMAS simulation in the bottom as well. Ill update this tmrw morning to have more PUMAS years.

**Comment by islasimpson** [comment]
[Here](https://www.cgd.ucar.edu/staff/islas/watervapor/taperecorder_jan2022.pdf) are some tape recorders.

**Comment by islasimpson** [comment]
Also, someone sent me these papers the other day.  They discuss how better resolving tropical waves in the TTL can lead to dehydration of the stratosphere because the waves have warm and cold regions and the cold regions control how much gets into the stratosphere.  I haven't had a chance to read them yet, but I wonder if there are some diagnostics in there that we could use to test if that's what's happening.  If it is, then maybe we're barking up the wrong tree if we think it's physics that's responsible.

https://acp.copernicus.org/articles/15/3517/2015/
https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2014GL062737
https://acp.copernicus.org/articles/21/2469/2021/

**Reply by adamrher** [reply]
Thanks. We could re-create their figure two (in the second paper) during the wet phase.

> maybe we're barking up the wrong tree if we think it's physics that's responsible.

I've been arguing for a while it's the dynamics, not the physics. This was the point of the DQCORE diagnostics. I don't think my understanding is inconsistent with the mechanisms discussed in these papers.

**Comment by adamrher** [comment]
@dan800 copying your post into this discussion on the impacts of the "sc" compset modifier.

> @JulioTBacmeister @adamrher @islasimpson: So I had a look at a FWscHIST.ne30_L83 case to see what's in the name list and chemistry. I think this is at least part of how it's set up in 'sc'
> 
> Molecular diffusion of species is off:
> do_molec_press = 0.1D0 and the top level is 0.0081 hPa (0.8Pa)
> 
> non-LTE cooling calcs are on (I think) and use constituents in the WACCM forcing file as part of the calculation - CO2, O, O2 and O3 if I recall correctly.
> 
> The only prognostic constituents are H2O, CH4, N2O, CFC11, CFC12, H2O2, H2SO4, SO2 & DMS. The chemistry (if you can call it that) for CH4, N2O, CFC11 and CFC12 is just a simple loss mechanism - water is produced from the loss of CH4.
> DMS is lost with reactions with OH and NO3 (both specfied) to produce SO2 which then reacts with OH to produce H2SO4. Not sure what H2O2 does in the model - probably something to do with the aerosols (its reaction with OH is a source of water but I doubt it's significant).
> 
> Looks like there's 20 aerosol species in total.
> 
> So not sure what switching off SC for a low top model gets you unless it drops some of these tracers. I can't imagine reading in a heating rate and calling Fomichev cost 10% of the model. But we can certainly see if turning off those two things really changes the 50 to 80 km region much. If it doesn't we can just ditch it for all CAM configurations.

**Reply by islasimpson** [reply]
Here is a plot of H2O.  It's comparing WACCM with with L83, both coupled piControl, FV.  It's a bit hard to separate out any influence of the upper boundary condition from the overall dry bias.  @dan800 - let me know if there are other diagnostics that would show this better.  I don't think there's a rush - we maybe don't care about this too much for the Scripps seasonal prediction project as they are short runs, initialized from obs.

![H2O](https://user-images.githubusercontent.com/41842751/152901380-2a6bf5d6-94dc-4cd3-a229-edeef2bfa06a.jpeg)

**Reply by dan800** [reply]
@islasimpson Can you check the value at the top of the model in L83? Is it setting it to the value used in WACCM (i.e., 20 ppb)? 
```
     subroutine set_tgcm_ubc( lchunk, ncol, mmr, mw_dry )
        !--------------------------------------------------------------------
        !       ... Set the upper boundary values h2o, h2, and h
        !--------------------------------------------------------------------
         real(r8), parameter ::  h2o_ubc_vmr = 2.e-8_r8            ! fixed ub h2o concentration (kg/kg)
```

**Reply by islasimpson** [reply]
I can look at the top lev in the netcdf files but I guess that's not the top.   Here's a snapshot of the top lev.  Minimum values here are higher than that (on the order 5e-7).  Is this helping or do you mean actually figuring it out from the code?

![Screenshot from 2022-02-08 18-27-30](https://user-images.githubusercontent.com/41842751/153104672-256c099a-1021-4150-9e80-43295d0ff72f.png)

**Comment by adamrher** [comment]
Here's some more analysis of the differences between FHIST and FWscHIST. Timings (core hrs per sim yr) using 2100 tasks
```
FWscHIST: 7364.40, 7333.62
FHIST: 6121.43, 5976.72
```

Both runs have lots of high frequency output ... and all of the fincls have interpolate_output=T. It is surprising to me how expensive these sims are ... I have FHIST L58 runs w/ more modest i/o and the cost from one timing file is 4680.03 on 1800 tasks. That's much closer to the expected incr. in cost going from L32->L58. 

Returning to the matter at hand, there's no denying that FWscHIST is a robust 20% increase in cost over FHIST. That's nothing to balk at. 20% is equal to the cost savings of the pg2 grid.

The reason it's more expensive is probably because it has more tracers. FHIST has 35 tracers, FWscHIST has 43. (@PeterHjortLauritzen any idea of whether the +20% is consistent w/ a back-of-the-envelope calc due to 8 more tracers?). The additional tracers are: 

    12  CH4       CH4                                                                                                                               dry
    13  N2O       N2O                                                                                                                               dry
    14  CFC11     CFC11                                                                                                                             dry
    15  CFC12     CFC12                                                                                                                             dry
    40  AOA1      Age-of_air tracer 1                                                                                                               wet
    41  AOA2      Age-of_air tracer 2                                                                                                               wet
    42  HORZ      horizontal tracer                                                                                                                 wet
    43  VERT      vertical tracer                                                                                                                   wet

Well the first four seem kind of important. How were these treated in prior low-top climate sensitivity runs? I can see the age of air important for diagnostic purposes. What are these HORZ and VERT tracers?

If there are other causes of the increase, maybe someone can glean some info from looking at a diff of the atm_in files between these two compsets. Execute this command on cheyenne:
```
diff /glade/scratch/aherring/cam6_3_043_FWscHIST_ne30pg3_ne30pg3_mg17_mct_600pes_220120_camdev_dynlb0/run/atm_in /glade/scratch/aherring/cam.clubbmf_FHIST_ne30pg3_ne30pg3_mg17_L58camdev_1800pes_220108_3macmic_Nx2yrs/run/atm_in
```

Lastly, I want to illustrate the main climate difference between these two configuration. There is a modest moistening of the lower stratosphere in FHIST compared to FWscHIST ([here](https://webext.cgd.ucar.edu/FSCAM/FWscHIST/L32_REF/02042022/temp_dzonal.Q.logp.scsens.pdf)). But I think the temperature signal is more interesting, with the Tropical cold point warmer [by about 1K in FHIST](https://webext.cgd.ucar.edu/FSCAM/FWscHIST/L32_REF/02042022/temp_dzonal.T.logp.scsens.pdf). Any ideas why?

**Reply by dan800** [reply]
@adamrher The 'modest moistening of the lower strat' is odd considering CH4 is a source of water vapor in the stratosphere and I think you are saying CH4 is absent in FHIST. I am guessing the 1K diff is the main cause. There's several papers on the impact of resolution and how ozone is specified that can cause large differences in the water vapor entering the stratosphere. Are you using ozone from a run with chemistry on the same vertical grid in each of these runs. You need a consistent profile otherwise you will not put the SW heating at the right location.

IF the plan is to always run a model with chemistry for each required SSP then maybe you can get away with specifying the CFCs - not sure.

The dynamical tracers are likely going to be required for delivered simulations (@islasimpson can comment hopefully) - they are omitted in FHIST probably because it will highlight just how bad the low top is in the stratosphere! IIRC there's a simple namelist to switch them off and on - so for tuning you can likely have them off no matter the model top. AOA is useless unless your run is a few decades long - it takes 7 or more years for the circulation to spin up the tracer.

If you are really searching for further efficiencies in model tuning for short runs, maybe looking at the 20 aerosols would be an area to investigate.

**Reply by dan800** [reply]
I had a quick look at the namelist definitions - toggling age of air tracers seems to be possible at configure:

aoa_tracers_flag | aoa_tracers_nl | test_tracers | logical
-- | -- | -- | --
Valid Values ['.true.', '.false.']Possible Default ValuesDescription and out-of-the-box DefaultIf true age of air tracers are included.  This variable should not be set by the user.  It will be set by build-namelist to be consistent with the '-age_of_air_trcs' argument specified to configure. Default: set by configure

**Reply by dan800** [reply]
Relevant papers: 
Novack showed massive differences between an interactive chemistry and specified in HadGEM
doi:[10.1038/nclimate2451](https://doi.org/10.1038/nclimate2451).
I didn't in WACCM:
https://doi.org/10.1002/2016GL068344
Other groups were in between.
It does seem it is partly vertical resolution dependent -  In HadGEM they move the ozone profile up and down with the tropopause to fix this issue
https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2019MS001714

**Reply by islasimpson** [reply]
@PeterHjortLauritzen might be better placed to comment on the tracers.  I'm not sure what the horizontal and the vertical ones are, but I agree that for any L93 simulations, we'd probably want to be producing the age of air for simulations that are being delivered to the community.

**Reply by dan800** [reply]
From an email in 2013:
Dan Marsh <MARSH@ucar.edu>
Mar 13, 2013, 10:12 AM
to Rolando, Doug, Anne, Jean-Francois, vyudin, fvitt

I agree.    I'd go further.  If they are not used, I'd suggest deleting them and giving them names
that reflect what they are supposed to do.  Please make sure the metadata (long_name, etc.)
also reflects what they are exactly.  And, since they will have limited use, we would
need a good argument for having them as default output even in the h0s.

Dan

**Reply by dan800** [reply]
It's worrying you are not using a self consistent ozone profile to force the L58 model but a low res WACCM grid. SW heating rates will be off/inconsistent. A new prescribed ozone file needs to be made from a L58 run with chemistry.

&prescribed_ozone_nl
 prescribed_ozone_datapath              = '/glade/p/cesmdata/cseg/inputdata/atm/cam/ozone_strataero'
 prescribed_ozone_file          = 'ozone_strataero_WACCM_L70_zm5day_18500101-21010201_CMIP6histEnsAvg_SSP370_c190403.nc'
 prescribed_ozone_name          = 'O3'
 prescribed_ozone_type          = 'SERIAL'

**Reply by adamrher** [reply]
@dan800 I had a look at the ozone file being used in FHIST and FWscHIST. I tried to make it a raster plot the reflects underlying L70 grid the ozone file is on, for each of the two model runs. The model runs are L58. See here 1979-1998 Temperature and O3, with temperature being from the model runs and so is it's the L58 grid.

![temp_dzonal_T](https://user-images.githubusercontent.com/29961476/153089889-331ed397-f985-4ce8-a559-6b61e7b02c21.png)

It looks to me that L70 is higher resolution than L58 in most of the Tropical stratosphere. I agree that we should replace this ozone file from a L58 or L93 CHEM run before releasing ... but I think using the L70 dataset as we are developing the physics is probably OK. 

Should open another discussion "generating consistent chemistry datasets for L58/L93 specified chemistry configurations" so we don't forget to address this and other datasets we may want to produce? If we start opening discussion threads for every "to do" for CAM7, would it get out of control? Or is this a convenient way to manage our priorities?

Regarding the temperature changes between FHIST and FWscHIST. I should probably look at radiation tendencies. Maybe I should look at CFCs as well? Any suggestions are welcome!

**Reply by dan800** [reply]
I must admit I am not fully understanding what you are showing me here. It's the interpolation of ozone onto the new grid that worries me and differences in QRS between a model with specified ozone and prognostic ozone. I thought we had increased vertical resolution in the TTL compared to good old WACCM L70 in this new workhorse model. I also worry about using a zonal mean file - is the ozone at the cold point tropopause the same over the pacific as over the Tibetan plateau?

I am looking for physical consistency. Is it hard to run the current candidate workhorse model with chemistry? Why not save out the 3-D ozone and use that to feed RRTMG? Why would you want to wait until the release to do this? If we see differences between the cold point temperature between prescribed and calculated ozone then we may have to go down the route that HadGEM did of adjusting the ozone profile relative to the CPT.

**Reply by adamrher** [reply]
I misspoke when I said L58 and L93 are identical where they overlap. The top ten or so levels of L58 are coarsened relative to the equivalent heights in the L93 grid:
 
<img width="594" alt="Screen Shot 2022-02-08 at 5 45 04 PM" src="https://user-images.githubusercontent.com/29961476/153100745-98ebbad3-4aa2-4453-90e0-e1d717d9d69a.png">

I've no problem w/ reading in 3D ozone profiles. Not sure why a 2D file is used (maybe for space reasons? There's thousands of time samples in the files). But to be clear at what you're looking at, you are looking at the ozone data in the waccm file that is read in, and climo temperature of the L58 grid (which use the corresponding ozone data).

**Reply by dan800** [reply]
@adamrher @cecilehannay In looking at the runs listed in 'issues' I don't see any with either L58 or L93. What's the plan for switching over to these grids for CAM7 development?

**Reply by islasimpson** [reply]
In all of these runs, L58 is referred to as L48_BL10.  So all the L48_BL10 runs are using the L58 grid.

**Reply by adamrher** [reply]
you can click "labels" and select L58, L93, etc in the issues tab and it will filter the runs to ur liking

**Reply by dan800** [reply]
Ah - a little confusing! Since the grid is settled, maybe switch case names to L58 and L93?

**Reply by cecilehannay** [reply]
The label for the L58 is 'L58'
https://github.com/NCAR/amwg_dev/labels/L58

The label for the L93 is 'L93'
https://github.com/NCAR/amwg_dev/labels/L93

You can find the  labels at:
https://github.com/NCAR/amwg_dev/labels

![View labels](https://user-images.githubusercontent.com/9723220/153253937-acac0a34-eca2-415b-a5c8-809b60d34218.png)

As Isla explained these grids were called:
L48_BL10 and L83_L10
in the long names to differentiate the runs that were done with extra levels in the PBL when we were developing the grid.

Note that all the runs are not available on the amwg_dev github. (Older runs before we created the github are not all there).

**Reply by cecilehannay** [reply]
> Ah - a little confusing! Since the grid is settled, maybe switch case names to L58 and L93?

@dan800: We could change the casenames now the grids are settled. 

The reason we did that way was that when we were developing grids, we had a flurry of grids. It was confusing and we wanted to differentiate grids with the BL levels and without.

For instance, at some point, there was one **L91 without BL levels** and **L93 with BL levels**. 
So differentiate them, we called the grids:
**L91** without extra BL levels
**L83_BL10** with extra BL level

But I think it is fine to change the names now that all the grids have extra BL levels.

**Reply by adamrher** [reply]
@dan800 I think we should open up a new discussion about what tracers we want in "specified chemistry" runs. I think we can omit 6 for sure (HORZ, VERT, AOA1, AOA2, and the two graupel tracers). Do you want to open it up with your thoughts on what else to consider?

**Reply by dan800** [reply]
OK - I'd say the discussion is more 'Tracers in CAM7'. The hope is we unify physics and # of tracers for L93 and L58. There's got to be a really good reason to add complexity and overhead should these choices diverge.

**Comment by dan800** [comment]
Hi Isla,

I think you need to run WACCM on the same grid where they overlap to really get to the bottom of this. If the line commented out in Byron’s code means it it using the fixed boundary condition for waccm at 140 km at 80 km (ie practically zero) that certainly will yield low values higher up. Not convinced the differences in the UTLS are not just from differences in tropopause height. To help with the Scripps project just changing the UBC fixed value might be a big improvement, but for our needs we can do better - a seasonally varying value from WACCM. 

Regards,

Dan
